File: example/handler.py
import sys
sys.path.append('gen-py.tornado')

from thrift.protocol import TJSONProtocol
from scrapper import Scrapper
from thrift_amqp_tornado import TAMQPTornadoServer
from tornado import ioloop, gen
from tornado.httpclient import AsyncHTTPClient
http_client = AsyncHTTPClient()
loop = ioloop.IOLoop.instance()


@gen.engine
def fetch_data(url, callback):
    try:
        r = yield gen.Task(http_client.fetch, url)
        print "done"
        callback(Scrapper.Blog(url=url, content=r.body[:100]))
    except:
        print "Something went wrong"
        callback(Scrapper.Blog())


class ScraperHandler(object):
    def __init__(self):
        pass

    def scrape(self, url):
        return gen.Task(fetch_data, url)

handler = ScraperHandler()
processor = Scrapper.Processor(handler)
pfactory = TJSONProtocol.TJSONProtocolFactory()
server = TAMQPTornadoServer(processor, pfactory)

server.start()
loop.start()

File: example/handler.rb
$LOAD_PATH.unshift 'gen-rb'

require 'scrapper'
require 'thrift'
require 'thrift/amqp/ruby'

class ScrapHandler
  def scrape(url)
    Blog.new(url: url, content: "Hi")
  end
end

handler = ScrapHandler.new
processor = Scrapper::Processor.new(handler)
pfactory= Thrift::JsonProtocolFactory
server = Thrift::AmqpRpcServer.new(processor,
                                   host: 'localhost',
                                   port: '5672',
                                   exchange: 'rpc-server-exchange',
                                   queue_name: 'thrift',
                                   protocol_factory: pfactory)

server.serve
